integrationtests/as-lucene-directory/src/test/java/org/infinispan/test/integration/as/wildfly/InfinispanModuleMemberRegistrationIT.java

```diff
@@ -128,10 +128,52 @@ public void testNewMemberSearch() throws Exception {
    @Test
    @InSequence(value = 3)
    @OperateOnDeployment("dep.active-2")
+   public void testNewMemberLuceneSearch() throws Exception {
+      List<Member> search = memberRegistration.luceneSearch("Peter");
+
+      assertFalse("Expected at least one result after the indexing", search.isEmpty());
+      assertEquals("Lucene search hasn't found a member", "Peter O'Tall", search.get(0).getName());
+   }
+
+   @Test
+   @InSequence(value = 4)
+   @OperateOnDeployment("dep.active-2")
    public void testNonExistingMember() throws Exception {
       List<Member> search = memberRegistration.search("TotallyInventedName");
 
       assertNotNull("Search should never return null", search);
       assertTrue("Search results should be empty", search.isEmpty());
    }
+
+   @Test
+   @InSequence(value = 5)
+   @OperateOnDeployment("dep.active-2")
+   public void testLuceneNonExistingMember() throws Exception {
+      List<Member> search = memberRegistration.luceneSearch("TotallyInventedName");
+
+      assertNotNull("Search should never return null", search);
+      assertTrue("Search results should be empty", search.isEmpty());
+   }
+
+   @Test
+   @InSequence(value = 6)
+   @OperateOnDeployment("dep.active-2")
+   public void testPurgeIndex() throws Exception {
+      memberRegistration.purgeMemberIndex();
+      List<Member> search = memberRegistration.search("Peter");
+
+      assertNotNull("Search should never return null", search);
+      assertTrue("Search results should be empty", search.isEmpty());
+   }
+
+   @Test
+   @InSequence(value = 7)
+   @OperateOnDeployment("dep.active-2")
+   public void testReIndex() throws Exception {
```

Comment:
@gustavonalle you were spot-on on the issue, but @alanfx is adding new tests (wasn't me writing that).

@alanfx Be careful when extending these tests as it's a bit of a minefield: the setup and configuration of this test is very peculiar. There are at least two things which are very slippery (but were good enough for the purposes of my original test):

 - The two "nodes" are not sharing the same database: as it's two different JVMs and each one is starting its own in-memory H2 database instance. This means you can't really test that what node-1 is writing can be found by node-2 (or vice-versa), unless you carefull keep this in mind - for example using projections or resorting to direct SQL to replicate data, or start an actually shared database.

 - This test isn't using an actual backend, it's resorting to a time-shared write access using 'exclusive_index_use=false'. The JMS based integration tests should provide you with a more reliable foundation to play with all features w/o having to keep in mind the limitations - however for the purpose of testing the Infinsipan Directory integration this should be good enough. Just something to keep in mind.